src/trigger/progress/threshold/amount_type.rs

```diff
@@ -0,0 +1,16 @@
+use serde::{Deserialize, Serialize};
+use strum::{AsRefStr, EnumIter, IntoStaticStr, VariantArray};
+
+#[derive(Debug, Default, Clone, AsRefStr, IntoStaticStr, EnumIter, Serialize, Deserialize)]
+pub enum AmountType {
+    // Intensity.
+    #[default]
+    Intensity,
+
+    // Duration.
+    Duration,
+}
+
+impl VariantArray for AmountType {
```

Comment:
This can be auto generated via adding `VariantArray` to the `#[derive(...)]` attribute, no need to implement it manually.